### PR TITLE
feat: Poc/postgresql

### PR DESCRIPTION
**Pull request by GeorginaFraire**
#### What does this PR do?
- Migrates from MySQL to PostgreSQL
- update the Prisma schema 
- update the Components
- update the controller 
- update the tests
- add the files necessaries to create a new db in PostgreSQL and migrate the data. 

#### How should this be manually tested?
1. Include PostgreSQL in the docker container [the configuration is on **_docker-compose.yaml_**] 
           run de command `docker-compose up -d`
2. Download the file '_dump_wizeq_01_08_2023.sql_' from keybase
3. Run the command to access to Postgres 
 `docker-compose exec postgres psql -U postresql` 
 and create the database `CREATE DATASE wizeq`
4.  Copy the  `dump_wizeq_01_08_2023.sq` to the postgres container 
`docker cp  <REPLACE WITH YOUR PATH>/dump_wizeq_01_08_2023.sql postgres:/dump_wizeq_01_08_2023.sql
`
5. Verify that the file is inside the container 
`docker-compose exec postgres ls`

Screen Shot 2023-08-01 at 20 44 35

6.  run the command `docker-compose exec postgres psql -U postgres -d wizeq -a -f dump_wizeq_01_08_2023.sql` to copy the dump in your local DB.

7. Connect to the DB ` psql -h localhost -p 5431 -U postgres -d wizeq ` [use the password: wizeq_password]
8.  Verify that you have questions in your database with the following query: `SELECT COUNT (*) FROM wizeq.questions;`
9. Change the variable. `DATABASE_URL` in your .env file
`postgresql://USER:PASSWORD@localhost:PORT/DB?schema=SCHEMA`
You can send me a DM to share with you the values. 
11.  Run the project `npm run dev`
--- TESTING ---
1. Verify that the container for testing '**_integration-tests-prisma_**' is running `docker ps `
2. run the command `npm run test:integration` 
3. 
Screen Shot 2023-08-03 at 12 03 06

# 👁️ 
if you want to access to the DB in Postgres use ` psql -h localhost -p 5431 -U postgres -d wizeq `

# ⚠️ 
testing job is falling because is setting for MySQL.

### CODE REVIEW

**Comment by albertcastaned** [comment]
Amazing job @GeorginaFraire ! 

Two questions
1. For the sql dump is it straight from the existign database or some modifications were needed? To see if we can use GCP SQL migration directly or we would need some manual updates.
2. Does Prisma+Postgres only allow lowercases for the schema variables?

**Comment by GeorginaFraire** [comment]
> Amazing job @GeorginaFraire !
> 
> Two questions
> 
> 1. For the sql dump is it straight from the existign database or some modifications were needed? To see if we can use GCP SQL migration directly or we would need some manual updates.

The dump file of the result of the migration that I did was with pgloader and the Prisma shema.


> 2. Does Prisma+Postgres only allow lowercases for the schema variables? 
It allows to create the database using the schema that we already have, but when I tried to do the migration I got theses errors.

**Comment by albertcastaned** [comment]
> @albertcastaned
> 
> > Amazing job @GeorginaFraire !
> > Two questions
> > 
> > 1. For the sql dump is it straight from the existign database or some modifications were needed? To see if we can use GCP SQL migration directly or we would need some manual updates.
> 
> The dump file of the result of the migration that I did was with pgloader and the Prisma shema.
> 
> > 2. Does Prisma+Postgres only allow lowercases for the schema variables?
> 
> It allows to create the database using the schema that we already have, but when I tried to do the migration I got theses errors.

@GeorginaFraire Nice, so in develop and production we would need to run pgloader script to get a equivalent sql dump?  Not sure if GCP SQL converter would work 🤔

**Comment by albertcastaned** [comment]
The integration tests shouldn't be too difficult fixing, I can do that if that's okay.

File: config/mysql/my.cnf
Comment (Carlos-Reyna) [inline config/mysql/my.cnf:1]: Is this file related to pgloader?  I do have a hard time finding the use of this file 😅

**Review by Carlos-Reyna — APPROVED** [review]

**Comment by jackbravo** [comment]
What is that `GCP SQL converter` tool that you mention @albertcastaned ? Sounds interesting.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -10,23 +10,27 @@ jobs:
     strategy:
       matrix:
         node-version: [14.x]
-    
+    services:
```
Comment (jackbravo) [inline .github/workflows/test.yaml:13]: There is also a `harmon758/postgresql-action@v1` that would be used the same way as the MySQL one. I wonder which option (a service with a postgres image, or a postgres-action) is faster.

This is an example of how to use it: https://github.com/wizeline/remix-project-lab/blob/main/.github/workflows/deploy.yml#L81

**Review by jackbravo — APPROVED** [review]
If we are changing to lower case to make it easier to write PostgreSQL queries, I would have preferred to use `_` to separate, so `updatedAt` would be changed to `updated_at`. Postgres also supports mixed case, but you need to quote column names that are mixed case

This is just a cosmetic suggestion.

**Comment by albertcastaned** [comment]
> What is that `GCP SQL converter` tool that you mention @albertcastaned ? Sounds interesting.

Did a quick search some weeks ago and found this https://cloud.google.com/blog/products/databases/database-cloud-migrations-made-easy-for-postgresql-mysql. However, looking closer it looks like it is not what I thought it was (mysql --> postgres) 😢

File: .github/workflows/test.yaml (hunk 1 above)
Comment (albertcastaned) [reply .github/workflows/test.yaml:13]: > There is also a `harmon758/postgresql-action@v1` that would be used the same way as the MySQL one. I wonder which option (a service with a postgres image, or a postgres-action) is faster.
> 
> This is an example of how to use it: https://github.com/wizeline/remix-project-lab/blob/main/.github/workflows/deploy.yml#L81

Oh nice, didn't know about this action. I ended up using a service in this other PR https://github.com/wizeline/wize-q-remix/pull/259. We are investigating the test failures, but the database connection works. We can look in the future if it's faster to use action.

**Comment by albertcastaned** [comment]
> If we are changing to lower case to make it easier to write PostgreSQL queries, I would have preferred to use `_` to separate, so `updatedAt` would be changed to `updated_at`. Postgres also supports mixed case, but you need to quote column names that are mixed case
> 
> This is just a cosmetic suggestion.

Agreed, this would be a cleaner solution, although it would also mean updating the SQL dump also. Do you think this would be possible / easy to do ? @GeorginaFraire

**Comment by albertcastaned** [comment]
I think we can merge this now @GeorginaFraire 👍 . Just a question, should we keep the mysql config files + scripts in source control?

File: config/mysql/my.cnf
Comment (GeorginaFraire) [reply config/mysql/my.cnf:1]: yes. I'll remove it

**Comment by GeorginaFraire** [comment]
> > If we are changing to lower case to make it easier to write PostgreSQL queries, I would have preferred to use `_` to separate, so `updatedAt` would be changed to `updated_at`. Postgres also supports mixed case, but you need to quote column names that are mixed case
> > This is just a cosmetic suggestion.
> 
> Agreed, this would be a cleaner solution, although it would also mean updating the SQL dump also. Do you think this would be possible / easy to do ? @GeorginaFraire

I'm no sure if it's a good idea. I dont know how prisma and the DB are going to react, maybe it will ask to reset the data.

**Comment by GeorginaFraire** [comment]
> I think we can merge this now @GeorginaFraire 👍 . Just a question, should we keep the mysql config files + scripts in source control?

Yes, we can remove them.

**Pull request merged**